src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorParser.cs

```diff
@@ -58,43 +61,58 @@ public class MvcRazorParser : RazorParser
+            var chunksFromGlobalImports = inheritedCodeTrees
+                .Reverse()
+                .SelectMany(tree => tree.Chunks);
+            var chunksInOrder = defaultInheritedChunks.Concat(chunksFromGlobalImports);
             foreach (var chunk in chunksInOrder)
             {
-                var addHelperChunk = chunk as AddTagHelperChunk;
-                if (addHelperChunk != null)
+                if (chunk is AddTagHelperChunk)
                 {
-                    var descriptor = new TagHelperDirectiveDescriptor(addHelperChunk.LookupText,
-                                                                      SourceLocation.Undefined,
-                                                                      TagHelperDirectiveType.AddTagHelper);
+                    var descriptor = CreateInheritedTagHelperDirectiveDescriptor(
+                        ((AddTagHelperChunk)chunk).LookupText,
+                        TagHelperDirectiveType.AddTagHelper);
+
                     descriptors.Add(descriptor);
                 }
-                else
+                else if (chunk is RemoveTagHelperChunk)
                 {
-                    var removeHelperChunk = chunk as RemoveTagHelperChunk;
-                    if (removeHelperChunk != null)
-                    {
-                        var descriptor = new TagHelperDirectiveDescriptor(removeHelperChunk.LookupText,
-                                                                          SourceLocation.Undefined,
-                                                                          TagHelperDirectiveType.RemoveTagHelper);
-                        descriptors.Add(descriptor);
-                    }
+                    var descriptor = CreateInheritedTagHelperDirectiveDescriptor(
+                        ((RemoveTagHelperChunk)chunk).LookupText,
+                        TagHelperDirectiveType.RemoveTagHelper);
+
+                    descriptors.Add(descriptor);
+                }
+                else if (chunk is TagHelperPrefixDirectiveChunk)
+                {
+                    var descriptor = CreateInheritedTagHelperDirectiveDescriptor(
+                        ((TagHelperPrefixDirectiveChunk)chunk).Prefix,
+                        TagHelperDirectiveType.TagHelperPrefix);
+
+                    descriptors.Add(descriptor);
                 }
             }
 
             return descriptors;
         }
 
-        private class GlobalImportAddRemoveTagHelperVisitor : TagHelperDirectiveSpanVisitor
+        private static TagHelperDirectiveDescriptor CreateInheritedTagHelperDirectiveDescriptor(
+            string directiveText,
+            TagHelperDirectiveType directiveType)
+        {
+            // Inherited directive descriptors have undefined source locations because the source that created them
+            // is not in the same file.
+            return new TagHelperDirectiveDescriptor(directiveText, SourceLocation.Undefined, directiveType);
+        }
```

Comment:
nit: this one-line method doesn't aid reader much and could be inlined as it was before